src/main/java/com/fasterxml/jackson/dataformat/xml/ser/XmlSerializerProvider.java

```diff
@@ -107,7 +109,8 @@ public void serializeValue(JsonGenerator gen, Object value, JavaType rootType,
             _initWithRootName(xgen, rootName);
             asArray = TypeUtil.isIndexedType(rootType);
             if (asArray) {
-                _startRootArray(xgen, rootName);
+                String indexedRootName = _rootNameLookup.findWrapperForIndexedType(getTypeOfCollection(rootType), _config);
```

Comment:
// May be `value` is right candidate as tested in spring with v2.9.9
String indexedRootName = _rootNameLookup.findWrapperForIndexedType(getTypeOfCollection(value), _config);